### PR TITLE
feat: updated compatibilities for istio

### PR DESCRIPTION
**Pull request by smyja**
## Summary
added more compatibilities and their non compatible versions as seen here https://istio.io/v1.12/docs/releases/supported-releases/#support-status-of-istio-releases

## Checklist

- [ ] If required, I have updated the Plural documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] I have added a meaningful title and summary to convey the impact of this PR to a user.
- [ ] I have added relevant labels to this PR to help with categorization for release notes.

### CODE REVIEW

File: static/compatibilities/istio.yaml [hunk 1]
```diff
@@ -33,4 +33,28 @@ versions:
 - version: 1.13.0
   kube: ['1.23', '1.22', '1.21', '1.20']
   requirements: []
-  incompatibilities: []
\ No newline at end of file
+  incompatibilities: []
+- version: 1.12
+  kube: ['1.22', '1.21', '1.20', '1.19']
+  requirements: []
+  incompatibilities: ['1.18', '1.17', '1.16']
```
Comment (michaeljguarino) [inline static/compatibilities/istio.yaml:40]: This is useful, but the data structure is actually a bit different, eg each will be a:

```yaml
name: <component-name>
version: <vsn>
```

struct. In this case you could use:

```yaml
name: kubernetes
version: 1.22
```

Comment (michaeljguarino) [reply static/compatibilities/istio.yaml:40]: The idea is this would need to be flexible to express incompatibilities between kubernetes operators as well

Comment (smyja) [reply static/compatibilities/istio.yaml]: > This is useful, but the data structure is actually a bit different, eg each incompatibility entry will be a:
> 
> ```yaml
> name: <component-name>
> version: <vsn>
> ```
> 
> struct. In this case you could use:
> 
> ```yaml
> name: kubernetes
> version: 1.22
> ```

okay, I've updated it. Kindly check to confirm if that's what you mean

**Review by michaeljguarino — APPROVED** [review]
lgtm

**Pull request merged**